### PR TITLE
Added documentation

### PR DESCRIPTION
**Pull request by MaxPostema**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] Clean commits
- [ ] No warnings during install
- [ ] Updated js typing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,71 @@
...
+```
+
+Every filter will have the following global properties:
+
+| name | required | description | 
+|------|----------|-------------|
+| name | true | string: name and id of the filter |
+| label | false | string: Name of the filter as used by the UI |
+| description | false |Descriptive text that is used by the UI |
+| initiallyCollapsed | true | boolean: Open/Closed state of the card that holds the filter |
```
Comment (connoratrug) [inline README.md]: Suggest using 'collapsed' , with required false and default of false

File: README.md [hunk 2]
```diff
@@ -1,5 +1,71 @@
...
+
+Every filter will have the following global properties:
+
+| name        | required | default value | description | 
+|-------------|----------|---------------|-------------|
+| name        | true     | nvt           | string: name and id of the filter |
+| label       | false    |               | string: Name of the filter as used by the UI |
+| description | false    |               | Descriptive text that is used by the UI |
+| collapsed   | false    | true          | boolean: Open/Closed state of the card that holds the filter |
+| type        | true     | nvt           | Vue Object that renders the filter |
```
Comment (connoratrug) [inline README.md]: maybe add a collapsable prop ( defaults to true ? ) 
maybe add a bulk selectable ( default true ?) or is this part of the Type being passed ?

Comment (fdlk) [reply README.md]: Let's first document what we have

File: src/components/FilterCard.vue [hunk 3]
```diff
@@ -3,9 +3,9 @@
     no-body
     class="filter-card mb-2"
   >
-    <b-card-header @click="collapsed = !collapsed">
+    <b-card-header @click="initiallyCollapsed = !initiallyCollapsed">
       <i
-        v-if="collapsed"
+        v-if="initiallyCollapsed"
```
Comment (fdlk) [inline src/components/FilterCard.vue]: I don't understand this.

File: README.md [hunk 4]
```diff
@@ -1,5 +1,71 @@
...
+```
+
+Every filter will have the following global properties:
+
+| name        | required | default value | description | 
+|-------------|----------|---------------|-------------|
+| name        | true     | nvt           | string: name and id of the filter |
+| label       | false    |               | string: Name of the filter as used by the UI |
+| description | false    |               | Descriptive text that is used by the UI |
+| collapsed   | false    | true          | boolean: Open/Closed state of the card that holds the filter |
```
Comment (fdlk) [inline README.md]: If we want a prop to control the state, we'll also need to fire an event when it changes. But I'd rather first describe what we have

File: README.md [hunk 5]
```diff
@@ -1,5 +1,71 @@
...
+| type        | true     | nvt           | Vue Object that renders the filter |
+
+The `type` field takes a VUE object that renders a filter. Note that every object can have additional filter properties
+
+### Additional filter properties
+#### CheckboxFilter 
+`options`: Describe the checkboxes. The value property is used as key and the text property is used as label by the UI
+```
+[{ value: 'value1', text: 'Checkbox 1' }, { value: 'value2', text: 'Checkbox 2' }]
```
Comment (connoratrug) [inline README.md]: what a about an id or key for when two options have the same value but a differ in text

**Comment by MaxPostema** [comment]
Closing PR, it is split into
#7 and #8

**Pull request closed without merging**